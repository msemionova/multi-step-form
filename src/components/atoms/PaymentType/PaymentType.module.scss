.payment-type {
  height: 4.8rem;
  background-color: var(--color-gray-background);
  border-radius: var(--border-radius-medium);
  display: flex;
  align-items: center;
  justify-content: center;
  gap: 1.6rem;

  h4 {
    color: var(--color-gray-text-light);
    font: var(--typography-input-text);
    transition: color 500ms;

    &.active {
      color: var(--color-black-text);
    }
  }

  .switch {
    position: relative;
    display: inline-block;
    width: 38px;
    height: 20px;
  }

  .switch input {
    display: none;
  }

  .slider {
    position: absolute;
    cursor: pointer;
    top: 0;
    left: 0;
    right: 0;
    bottom: 0;
    background-color: var(--color-gray-dark);
    -webkit-transition: .4s;
    transition: .4s;
    border-radius: var(--border-radius-round);
  }

  .slider:before {
    position: absolute;
    content: '';
    height: 14px;
    width: 14px;
    left: 3px;
    bottom: 3px;
    background-color: white;
    -webkit-transition: .4s;
    transition: .4s;
    border-radius: var(--border-radius-round);
  }

  input:checked + .slider {
    background-color: var(--color-primary);
  }

  input:checked + .slider:before {
    -webkit-transform: translateX(18px);
    -ms-transform: translateX(18px);
    transform: translateX(18px);
  }
}
