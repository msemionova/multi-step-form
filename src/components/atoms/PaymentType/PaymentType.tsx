import { useState } from 'react';
import styles from './PaymentType.module.scss';

export default function PaymentType() {
  const [isMonthly, setIsMonthly] = useState(true);

  return (
    <div className={styles['payment-type']}>
      <h4 className={isMonthly ? styles['active'] : ''}>Monthly</h4>
      <label className={styles['switch']}>
        <input
          type='checkbox'
          onChange={() => setIsMonthly(!isMonthly)} defaultChecked={!isMonthly} />
        <span className={styles['slider']}></span>
      </label>
      <h4 className={!isMonthly ? styles['active'] : ''}>Yearly</h4>
    </div>
  );
}
