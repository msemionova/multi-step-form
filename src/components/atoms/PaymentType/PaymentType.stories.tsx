import type { Meta, StoryObj } from '@storybook/react';

import PaymentType from './PaymentType';

const meta: Meta<typeof PaymentType> = {
  component: PaymentType,
};

export default meta;
type Story = StoryObj<typeof PaymentType>;

export const Default: Story = {
  args: {}
};
