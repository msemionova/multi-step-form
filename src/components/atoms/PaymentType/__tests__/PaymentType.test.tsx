import renderer from 'react-test-renderer';
import '@testing-library/jest-dom';

import PaymentType from '../PaymentType';

describe('<PaymentType />', () => {
  it('renders correctly', () => {
    const tree = renderer
      .create(<PaymentType />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
