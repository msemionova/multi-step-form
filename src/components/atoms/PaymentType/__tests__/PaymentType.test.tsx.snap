// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PaymentType /> renders correctly 1`] = `
<div
  className="payment-type"
>
  <h4
    className="active"
  >
    Monthly
  </h4>
  <label
    className="switch"
  >
    <input
      defaultChecked={false}
      onChange={[Function]}
      type="checkbox"
    />
    <span
      className="slider"
    />
  </label>
  <h4
    className=""
  >
    Yearly
  </h4>
</div>
`;
